layer/decorators/layer_wrapper.py

```diff
@@ -66,12 +66,29 @@ def __init__(
                     )
         self.layer.set_dependencies(paths)
 
-    def get_definition(self) -> FunctionDefinition:
+        self.args: Sequence[Any] = tuple()
+        self.kwargs: Mapping[str, Any] = {}
+
+    def bind(self, *args: Any, **kwargs: Any) -> Any:
+        self.args = args
+        self.kwargs = kwargs
+        return self
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        definition = self.get_definition(args, kwargs)
+        runner = definition.runner_function()
+        return runner()
+
+    def get_definition(
+        self, args: Sequence[Any], kwargs: Mapping[str, Any]
+    ) -> FunctionDefinition:
         self.layer.validate()
         current_project_full_name = get_current_project_full_name()
 
         return FunctionDefinition(
             func=self.__wrapped__,
+            args=args,
+            kwargs=kwargs,
```

Comment:
i find these confusing

Reply:
this is to support https://github.com/layerai/roadmap/issues/16. if args and kwargs are bound to the function via `func.bind(my_arg)`, then these are saved on the function definition and sent to the backend to be used during execution.

Reply:
We have "runtimes" for this purpose defining different logic depending on where the code runs. 

The code itself is only defined once by the user, so it's captured in a single class called FunctionDefinition. The same code for function and a generic wrapper (chosen depending on `@dataset` or `@model`) is the core that doesn't change, regardless of runtime.

Reply:
ok but nonetheless here you described that args/kwargs are needed for remote execution only? or i misunderstood

hence also my comment
> i understand better now. do they have to have the same name as when we use *args: Any, **kwargs: Any in signatures or can we call them remove_execution_args, remove_execution_kwargs?

maybe we dont have a choice but to have them inside the core for binding, but then we can be explicit it is not needed in local exec?

Reply:
sorry I didn't address that part, they are not just for remote, they are also for local execution.

Reply:
before this PR, local execution with args worked but remote didn't, because they were using two different code paths. a big reason for this PR is to simplify this code and have them both use the same code paths for the same logic. so this args/kwargs will be used in both scenarios.